src/validation.test.ts

```diff
@@ -617,4 +617,274 @@ describe('OpenAPIValidator', () => {
+        {
+          'X-Other-Integer': '42',
+          'X-Other-String': 42,
+          'X-Other-Boolean': 'true',
+        },
+        'createPet',
+        404,
+        'exact',
+      );
+      expect(valid.errors).toBeFalsy();
+    });
+
+    test('passes validation with additional headers and setMatchType is superset', async () => {
+      const valid = validator.validateResponseHeaders(
+        {
+          'X-Integer': '42',
+          'X-String': 42,
+          'X-Boolean': 'true',
+          'X-Other': 'false',
+        },
+        'listPets',
+        200,
+        'superset',
+      );
+      expect(valid.errors).toBeFalsy();
+    });
+
+    test('fails validation with missing headers and setMatchType is superset', async () => {
+      const valid = validator.validateResponseHeaders(
+        {
+          'X-Integer': '42',
+          'X-String': 42,
+        },
+        'listPets',
+        200,
+        'superset',
+      );
+      expect(valid.errors).toBeTruthy();
+    });
+
+    test('passes validation with missing headers and setMatchType is subset', async () => {
+      const valid = validator.validateResponseHeaders(
+        {
+          'X-Integer': '42',
+          'X-String': 42,
+        },
+        'listPets',
+        200,
+        'subset',
+      );
+      expect(valid.errors).toBeFalsy();
+    });
+
+    test('fails validation with additional headers and setMatchType is subset', async () => {
+      const valid = validator.validateResponseHeaders(
+        {
+          'X-Integer': '42',
+          'X-String': 42,
+          'X-Boolean': 'true',
+          'X-Other': 'false',
+        },
+        'listPets',
+        200,
+        'subset',
+      );
+      expect(valid.errors).toBeTruthy();
+    });
+
+    test('passes validation with any header set and setMatchType is any', async () => {
+      const valid = validator.validateResponseHeaders(
+        {
+          'X-Integer': '42',
+          'X-String': 42,
+        },
+        'listPets',
+        200,
+        'any',
+      );
+      expect(valid.errors).toBeFalsy();
+    });
+
+    test('fails validation with an invalid response object', async () => {
+      const valid = validator.validateResponseHeaders(
+        {
+          unknown: 'property',
+        },
+        'listPets',
+        200,
+        'exact',
+      );
+      expect(valid.errors).toBeTruthy();
+    });
+
+    test('fails validation with a missing response object', async () => {
+      const valid = validator.validateResponseHeaders(null, 'listPets', 200, 'exact');
+      expect(valid.errors).toBeTruthy();
+    });
+
+    test('passes validation with a res code handler fallback', async () => {
+      const valid = validator.validateResponseHeaders(
+        {
+          'X-Other-Integer': '42',
+          'X-Other-String': 42,
+          'X-Other-Boolean': 'true',
+        },
+        'listPets',
+        205,
+        'exact',
+      );
+      expect(valid.errors).toBeFalsy();
+    });
```

Comment:
Let's add a test to make sure header names are not case sensitive. I think right now they are, which according to the HTTP spec is incorrect.

https://stackoverflow.com/questions/5258977/are-http-headers-case-sensitive

Reply:
Nice catch. I'll add the test and the fix